contribs/gnogenesis/internal/txs/txs_add_packages_test.go

```diff
@@ -60,66 +64,236 @@ func TestGenesis_Txs_Add_Packages(t *testing.T) {
+		io.SetIn(
+			strings.NewReader(
+				fmt.Sprintf(
+					"%s\n",
+					password,
+				),
+			),
+		)
+
+		// Create the command
+		cmd := NewTxsCmd(io)
+		args := []string{
+			"add",
+			"packages",
+			"--genesis-path",
+			tempGenesis.Name(),
+			"--key-name",
+			keyname, // non-existent key name
+			"--gno-home",
+			keybaseDir, // temporaryDir for keybase
+			"--insecure-password-stdin",
+			dir,
+		}
+
+		// Run the command
+		cmdErr := cmd.ParseAndRun(context.Background(), args)
+		assert.Equal(t, cmdErr.Error(), "unable to sign txs, unable sign tx invalid account password")
+	})
+
+	t.Run("existent key correct password", func(t *testing.T) {
+		t.Parallel()
+
+		tempGenesis, cleanup := testutils.NewTestFile(t)
+		t.Cleanup(cleanup)
 
+		genesis := common.GetDefaultGenesis()
+		require.NoError(t, genesis.SaveAs(tempGenesis.Name()))
 		// Prepare the package
 		var (
 			packagePath = "gno.land/p/demo/cuttlas"
 			dir         = t.TempDir()
+			keybaseDir  = t.TempDir()
+			keyname     = "beep-boop"
+			password    = "somepass"
 		)
+		createValidFile(t, dir, packagePath)
+		// Create key
+		kb, err := keys.NewKeyBaseFromDir(keybaseDir)
+		require.NoError(t, err)
+		_, err = kb.CreateAccount(keyname, DefaultAccount_Seed, "", password, 0, 0)
+		require.NoError(t, err)
 
-		createFile := func(path, data string) {
-			file, err := os.Create(path)
-			require.NoError(t, err)
+		io := commands.NewTestIO()
+		io.SetIn(
+			strings.NewReader(
+				fmt.Sprintf(
+					"%s\n",
+					password,
+				),
+			),
+		)
 
-			_, err = file.WriteString(data)
-			require.NoError(t, err)
+		// Create the command
+		cmd := NewTxsCmd(io)
+		args := []string{
+			"add",
+			"packages",
+			"--genesis-path",
+			tempGenesis.Name(),
+			"--key-name",
+			keyname, // non-existent key name
+			"--gno-home",
+			keybaseDir, // temporaryDir for keybase
+			"--insecure-password-stdin",
+			dir,
 		}
 
-		// Create the gno.mod file
-		createFile(
-			filepath.Join(dir, "gno.mod"),
-			fmt.Sprintf("module %s\n", packagePath),
+		// Run the command
+		cmdErr := cmd.ParseAndRun(context.Background(), args)
+		require.NoError(t, cmdErr)
+
+		// Validate the transactions were written down
+		updatedGenesis, err := types.GenesisDocFromFile(tempGenesis.Name())
+		require.NoError(t, err)
+		require.NotNil(t, updatedGenesis.AppState)
+
+		// Fetch the state
+		state := updatedGenesis.AppState.(gnoland.GnoGenesisState)
+
+		require.Equal(t, 1, len(state.Txs))
+		require.Equal(t, 1, len(state.Txs[0].Tx.Msgs))
+
+		msgAddPkg, ok := state.Txs[0].Tx.Msgs[0].(vmm.MsgAddPackage)
+		require.True(t, ok)
+		require.Equal(t, "gpub1pgfj7ard9eg82cjtv4u4xetrwqer2dntxyfzxz3pq0skzdkmzu0r9h6gny6eg8c9dc303xrrudee6z4he4y7cs5rnjwmyf40yaj", state.Txs[0].Tx.Signatures[0].PubKey.String())
+		require.Equal(t, "cfe5a15d8def04cbdaf9d08e2511db7928152b26419c4577cbfa282c83118852411f3de5d045ce934555572c21bda8042ce5c64b793a01748e49cf2cff7c2983", hex.EncodeToString(state.Txs[0].Tx.Signatures[0].Signature))
```

Comment:
You can extract this as a constant in the top-level test wrapper (`TestGenesis_Txs_Add_Packages`)

Reply:
yes that's better 2a67769b4f165ffd4d9ed62cf27891387fa8b949